src/js/tabs/servos.js

```diff
@@ -1,36 +1,30 @@
 'use strict';
 
 TABS.servos = {};
-TABS.servos.initialize = function (callback) {
+TABS.servos.initialize = (callback) => {
 
     if (GUI.active_tab !== 'servos') {
         GUI.active_tab = 'servos';
     }
 
-    function get_servo_configurations() {
-        MSP.send_message(MSPCodes.MSP_SERVO_CONFIGURATIONS, false, false, get_servo_mix_rules);
-    }
-
-    function get_servo_mix_rules() {
-        MSP.send_message(MSPCodes.MSP_SERVO_MIX_RULES, false, false, get_rc_data);
-    }
-
-    function get_rc_data() {
-        MSP.send_message(MSPCodes.MSP_RC, false, false, get_boxnames_data);
-    }
-
-    function get_boxnames_data() {
-        MSP.send_message(MSPCodes.MSP_BOXNAMES, false, false, load_html);
-    }
-
     function load_html() {
         $('#content').load("./tabs/servos.html", process_html);
     }
-    get_servo_configurations();
+
+    Promise
+    .resolve(true)
+    .then(() => MSP.promise(MSPCodes.MSP_SERVO_CONFIGURATIONS))
+    .then(() => MSP.promise(MSPCodes.MSP_SERVO_MIX_RULES))
+    .then(() => MSP.promise(MSPCodes.MSP_RC))
+    .then(() => MSP.promise(MSPCodes.MSP_BOXNAMES))
+    .then(load_html)
+    .catch((error) => console.log(error.message));
```

Comment:
Do these need to be run in parallel or they can be in sequence?

If we make `initialise` async we could do something like for the sequential
```javascript
try {
  await MSP.promise(MSPCodes.MSP_SERVO_CONFIGURATIONS);
  await MSP.promise(MSPCodes.MSP_SERVO_MIX_RULES);
  await MSP.promise(MSPCodes.MSP_RC);
  await MSP.promise(MSPCodes.MSP_BOXNAMES);
  load_html();
} catch(error) {
  console.log(error)
}
```

Or for parallel
```javascript
try {
  await Promise.all([
    MSP.promise(MSPCodes.MSP_SERVO_CONFIGURATIONS),
    MSP.promise(MSPCodes.MSP_SERVO_MIX_RULES),
    MSP.promise(MSPCodes.MSP_RC),
    MSP.promise(MSPCodes.MSP_BOXNAMES),
  ])
  load_html();
} catch(error) {
  console.log(error)
}
```

Reply:
There is no particular sequence as they don't depend on each other. So the first solution would work but I'm in doubt in what advantage the parallel solution has over the former.

Reply:
Thanks @chmelevskij - it's working 😜 Updated now in `servos.js` will change in `configuration.js` and `motors.js` also

Reply:
Updated `motors.js` also. The initialize part is okay, but somehow the save function does not like the `async` approach. This snippet won't work even with `async function process_html() {` on line 222
```
$('a.save').on('click', async () => {
            // gather data that doesn't have automatic change event bound
            FC.MOTOR_CONFIG.minthrottle = parseInt($('input[name="minthrottle"]').val());
            FC.MOTOR_CONFIG.maxthrottle = parseInt($('input[name="maxthrottle"]').val());
            FC.MOTOR_CONFIG.mincommand = parseInt($('input[name="mincommand"]').val());

            if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_42)) {
                FC.MOTOR_CONFIG.motor_poles = parseInt($('input[name="motorPoles"]').val());
            }

            FC.MOTOR_3D_CONFIG.deadband3d_low = parseInt($('input[name="3ddeadbandlow"]').val());
            FC.MOTOR_3D_CONFIG.deadband3d_high = parseInt($('input[name="3ddeadbandhigh"]').val());
            FC.MOTOR_3D_CONFIG.neutral = parseInt($('input[name="3dneutral"]').val());

            FC.PID_ADVANCED_CONFIG.fast_pwm_protocol = parseInt(esc_protocol_e.val() - 1);
            FC.PID_ADVANCED_CONFIG.use_unsyncedPwm = $('input[id="unsyncedPWMSwitch"]').is(':checked') ? 1 : 0;
            FC.PID_ADVANCED_CONFIG.motor_pwm_rate = parseInt($('input[name="unsyncedpwmfreq"]').val());
            FC.PID_ADVANCED_CONFIG.digitalIdlePercent = parseFloat($('input[name="digitalIdlePercent"]').val());

            if (semver.gte(FC.CONFIG.apiVersion, "1.25.0") && semver.lt(FC.CONFIG.apiVersion, API_VERSION_1_41)) {
                FC.PID_ADVANCED_CONFIG.gyroUse32kHz = $('input[id="gyroUse32kHz"]').is(':checked') ? 1 : 0;
            }

            analytics.sendChangeEvents(analytics.EVENT_CATEGORIES.FLIGHT_CONTROLLER, self.analyticsChanges);
            self.analyticsChanges = {};

            try {
                await MSP.promise(MSPCodes.MSP_SET_ARMING_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_ARMING_CONFIG));
                await MSP.promise(MSPCodes.MSP_SET_MIXER_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_MIXER_CONFIG));
                await MSP.promise(MSPCodes.MSP_SET_MOTOR_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_MOTOR_CONFIG));
                await MSP.promise(MSPCodes.MSP_SET_MOTOR_3D_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_MOTOR_3D_CONFIG));
                await MSP.promise(MSPCodes.MSP_SET_ADVANCED_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_ADVANCED_CONFIG));
                await MSP.promise(MSPCodes.MSP_SET_FILTER_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_FILTER_CONFIG));
                await MSP.promise(MSPCodes.MSP_SET_FEATURE_CONFIG, mspHelper.crunch(MSPCodes.MSP_SET_FEATURE_CONFIG));
                await MSP.promise(MSPCodes.MSP_EEPROM_WRITE);
                GUI.log(i18n.getMessage('configurationEepromSaved'));

                GUI.tab_switch_cleanup(() => {
                    await MSP.promise(MSPCodes.MSP_SET_REBOOT);
                });

                analytics.sendChangeEvents(analytics.EVENT_CATEGORIES.FLIGHT_CONTROLLER, self.analyticsChanges);
                self.analyticsChanges = {};
                reinitialiseConnection();
            } catch (error) {
                console.log(error);
            }
        });
```
It produces the infamous
```
main.js:362 Uncaught TypeError: Cannot read property 'initialize' of undefined
    at main.js:362
    at Object.TABS.setup.cleanup (setup.js:351)
    at GuiControl.tab_switch_cleanup (gui.js:278)
    at HTMLAnchorElement.<anonymous> (main.js:266)
    at HTMLAnchorElement.dispatch (jquery.min.js:2)
    at HTMLAnchorElement.v.handle (jquery.min.js:2)
DevTools failed to load SourceMap: Could not load content for chrome-extension://ajdoecjhplhoahggfmgflfnlechcljoc/node_modules/jquery-textcomplete/dist/dist/jquery.textcomplete.min.map: System 
```

Reply:
If id doesn't matter about the sequence I'd use `Promise.all`, bunch of `await`s will block and wait for the previous to finish so you might get a bit of a lag if execution time adds up to 1s or smth

Reply:
We added some time ago one MSP command to execute various MSP commands at once. I don't remember the ID but can be useful for that.

Reply:
I found it: https://github.com/betaflight/betaflight-configurator/pull/1605

Reply:
MSP commands can definitely not be run in parallel - the firmware and the protocol lack the means to sequentialise / recombine responses.

Reply:
As for `MSP_MULTIPLE_MSP`, I'd be reluctant to use this as a wrapper to group multiple commands in configurator - this will break and require another rewrite as soon as one of the combined frames (request or response) exceeds 255 bytes in length, which will happen sooner or later due to the MSP protocol's nature of having commands grow in lenght with every change.